src/SFML/Audio/AudioDevice.cpp

```diff
@@ -502,39 +644,50 @@ bool AudioDevice::initialize()
                 err() << "Failed to read PCM frames from audio engine: " << ma_result_description(result) << std::endl;
         }
     };
+    playbackDeviceConfig.notificationCallback = [](const ma_device_notification* notification)
+    {
+        auto&                 notificationCallback = getNotificationCallback();
+        const std::lock_guard lock(notificationCallback.mutex);
+        const auto&           callback = notificationCallback.callback;
+
+        if (!callback)
+            return;
+
+        switch (notification->type)
+        {
+            case ma_device_notification_type_started:
+                callback(PlaybackDevice::Notification::DeviceStarted);
```

Comment:
Would it make sense and/or is it possible to also translate `notification->pDevice` and pass it to the callback?

Or is that easily determined with `sf::PlaybackDevice::getDevice()`?

Reply:
Some of these notifications come from the active backend itself, independent of what or whether there is still a device available. Even if there was information in `notification->pDevice` it wouldn't be reliable enough to base any algorithmic decisions on.